state/watcher.go

```diff
@@ -1645,8 +1645,20 @@ func (m *Machine) WatchHardwareCharacteristics() NotifyWatcher {
 }
 
 // WatchControllerInfo returns a NotifyWatcher for the controllers collection
-func (st *State) WatchControllerInfo() NotifyWatcher {
-	return newEntityWatcher(st, controllersC, modelGlobalKey)
+func (st *State) WatchControllerInfo() StringsWatcher {
+	return newCollectionWatcher(st, colWCfg{
```

Comment:
What does this change do? Is it that it used to be a model-specific collection and now it isn't, and the ids yielded from the watcher should still be model-local?

Reply:
Oh duh - missed that it had changed from a notify watcher to a strings watcher.

Reply:
This would be simpler if it didn't have to filter out other unrelated rows from the collection, wouldn't it?

Reply:
ok